Tests/UnitTests/Plugins/ActSVG/GridSvgConverterTests.cpp

```diff
@@ -0,0 +1,362 @@
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/ActSVG/GridSvgConverter.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+#include "Acts/Utilities/detail/Axis.hpp"
+#include "Acts/Utilities/detail/Grid.hpp"
+#include "actsvg/display/grids.hpp"
+
+#include <fstream>
+#include <memory>
+#include <string>
+#include <vector>
+
+using namespace Acts;
+using namespace Acts::detail;
+
+namespace {
+/// Helper method to turn a local bin into a string
+///
+/// @tparam local_bin_t the type for the local bin
+/// @param lBin the local bin to printed
+///
+/// @return a string for screen output
+template <typename local_bin_t, size_t DIM>
+std::string localToString(const local_bin_t& lBin) {
+  std::string lbString = "[";
+  for (size_t ib = 0; ib < DIM; ++ib) {
+    if (ib > 0u) {
+      lbString += std::string(", ");
+    }
+    lbString += std::to_string(lBin[ib]);
+  }
+  lbString += std::string("]");
+  return lbString;
+}
+}  // namespace
+
+BOOST_AUTO_TEST_SUITE(GridSvgConverter)
+
+BOOST_AUTO_TEST_CASE(BoundGridXY) {
+  using GlobalBin = size_t;
+  using LocalBin = std::array<size_t, 2u>;
+
+  // r-phi Axes & Grid
+  Axis<AxisType::Equidistant, AxisBoundaryType::Bound> axisX(-200., 200, 4);
+  Axis<AxisType::Equidistant, AxisBoundaryType::Bound> axisY(-200, 200, 6);
+  Grid<std::tuple<GlobalBin, LocalBin>, decltype(axisX), decltype(axisY)>
+      gridXY({axisX, axisY});
+
+  Svg::GridConverter::Options cOptions;
+  auto pGrid = Svg::GridConverter::convert(gridXY, {binX, binY}, cOptions);
+  pGrid._type = actsvg::proto::grid::type::e_x_y;
+
+  // Labelling the grid tiles
+  auto edgesX = axisX.getBinEdges();
+  auto edgesY = axisY.getBinEdges();
+
+  std::vector<actsvg::svg::object> targets = {};
+  size_t ig = 0;
+  for (auto [ix, x] : Acts::enumerate(edgesX)) {
+    if (ix > 0u) {
+      ActsScalar xp = 0.2 * edgesX[ix] + 0.8 * edgesX[ix - 1u];
+      for (auto [iy, y] : Acts::enumerate(edgesY)) {
+        if (iy > 0u) {
+          ActsScalar yp = 0.8 * edgesY[iy] + 0.2 * edgesY[iy - 1u];
+          decltype(gridXY)::point_t p = {xp, yp};
+          // Get local and global index
+          auto g = gridXY.globalBinFromPosition(p);
+          auto l = gridXY.localBinsFromPosition(p);
+          std::string gBin = std::string("g = ") + std::to_string(g);
+          std::string lBin =
+              std::string("l = ") +
+              localToString<decltype(gridXY)::index_t, decltype(gridXY)::DIM>(
+                  l);
+          std::vector<std::string> glBin = {gBin, lBin};
+          std::string gBinID = "g_" + std::to_string(ig++);
+          targets.push_back(
+              actsvg::draw::text(gBinID,
+                                 {static_cast<actsvg::scalar>(xp),
+                                  static_cast<actsvg::scalar>(yp)},
+                                 glBin));
+        }
+      }
+    }
+  }
+  pGrid._connections = targets;
```

Comment:
The grid doesn't really know about the targets, i.e. the draw objects you want to connect, it can draw the tiles etc, but you need to specify what to do with them.